enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,592 @@
+the current value of the `service.beta.kubernetes.io/aws-load-balancer-subnets` annotation.
+
+## Version Skew Strategy
+
+N/A
+
+## Operational Aspects of API Extensions
+
+N/A
+
+## Support Procedures
+
+### Multiple Subnets in Same AZ Support Procedure
+
+If a cluster admin has provided two subnets in the same Availability Zone (AZ), as
+discussed in [Invalid Subnet Annotation Values](#invalid-subnet-annotation-values),
+the IngressController with have a status with a relevant error:
+
+```yaml
+  - lastTransitionTime: "2024-04-02T22:07:59Z"
+    message: |-
+      The service-controller component is reporting SyncLoadBalancerFailed events like: Error syncing load balancer: failed to ensure load balancer: InvalidConfigurationRequest: ELB cannot be attached to multiple subnets in the same AZ.
+      The kube-controller-manager logs may contain more details.
+    reason: SyncLoadBalancerFailed
+    status: "False"
+    type: LoadBalancerReady
+```
+
+### Non-Existent Subnet Support Procedure
+
+If a cluster admin has provided a subnet that doesn't exist in the `Subnets` field,
+as discussed in [Invalid Subnet Annotation Values](#invalid-subnet-annotation-values),
+the IngressController with have a status with a relevant error:
+
+```yaml
+  - lastTransitionTime: "2024-04-02T22:07:59Z"
+    message: |-
+      The service-controller component is reporting SyncLoadBalancerFailed events like: Error syncing load balancer: failed to ensure load balancer: expected to find 1, but found 0 subnets
+      The kube-controller-manager logs may contain more details.
+    reason: SyncLoadBalancerFailed
+    status: "False"
+    type: LoadBalancerReady
+```
+
+## Alternatives
+
+### Automatically Recreating the Load Balancer
+
+One alternative to requiring cluster admins to recreate the LoadBalancer-type
+service, as described in [Effectuating Subnet Updates](#effectuating-subnet-updates),
+is to have the Ingress Operator automatically delete and recreate the service.
+
+While simpler in design, this approach could result in several minutes of unexpected
+disruption to ingress traffic. A cluster admin might not anticipate that updating
+the Subnets field could cause disruption, leading to an unwelcome surprise.
+
+### Immutability
+
+Another alternative is to make the `Subnets` field immutable. This would require a
+cluster admin to delete and recreate the IngressController if they wanted to update
+the subnets.
+
+The advantage of this approach is that the API Server would prevent any updates
+to the `Subnets` field, clearly indicating to the cluster admin that they would 
+need to recreate the IngressController. This is different from our current design,
+as described in [Effectuating Subnet Updates](#effectuating-subnet-updates),
+which is more subtle. With the current design, a cluster admin would need to check the
+IngressController's status after updating `Subnets` for instructions on how to proceed.
```

Comment:
Odds on an admin applying the change and not looking at the status, and therefore thinking that they've effected the change but actually have not?

Reply:
I don't know if I can answer. I personally would review the status after an update. This status also makes `Progressing=True` for the Ingress Cluster Operator. That provides greater visibility than only a condition on the IngressController alone.

@Miciah can you comment on whether we reasonably expect a cluster admin to review status after an IC API update?

Reply:
As Grant noted, the same workflow exists for mutating LB scope (internal/external), and the same risk exists that the cluster-admin will not think to check the ingresscontroller's status.  If we need to mitigate this risk, I suggest it be a separate enhancement.  (Maybe we just need an alert if the ingresscontroller has `Progressing=True` for more than a few minutes.)

Reply:
+1 for an alert if `Progressing=True` and the status message indicates an incomplete transaction.